lab3/euler.cpp

```diff
@@ -0,0 +1,44 @@
+#include <fstream>
+
+int TIMENUMBERS  = 100000;
+double x_0 = 5;
+double v_0 = 0;
+double dt = 0.01;
+double omega = 0.5;
+
+void output(double *x, double *x1, double *v, double *v1, double *t)
+{
+    std::ofstream foute;
+    foute.open("points.txt", std::ios::app);
+    foute << *x << '\t' << *x1 << '\t' << *v << '\t' << *v1 << '\t' << *t << std::endl;
+    foute.close();
+}
+
+
+
+int main()
+{
+    double x = x_0;
+    double x_i;
+    double v = v_0;
+    double x1 = x_0;
+    double x1_i, xz, vz;
+    double v1 = v_0;
+    double t = 0;
+    output(&x, &x1, &v, &v1, &t);
+    for(int i = 0; i < TIMENUMBERS; i++)
```

Comment:
How do we change solver for the very same equation  (houne instead of explicit euler)? How much code do we have to change? Is it maintainable that way? What if in our solver scheme we need more than single previous step (say to compute y[i+1] we need both y[i] and y[i-1]? BTW that's the naive solution - to represent y'' as y[i+1] - 2*[i] + y[i-1] and here we go: y[i+1] depends on two previous steps.

Reply:
OK, I got it, but I was too lazy to write it humanly, because I wanted to have more time.